libeos-payg/manager.c

```diff
@@ -796,8 +814,15 @@ epg_manager_add_code (EpgProvider   *provider,
   /* Reset the rate limiting history, since the code was successful. */
   clear_rate_limiting (self);
 
-  /* Kick off an asynchronous save. */
-  epg_manager_save_state_async (provider, self->cancellable, NULL, NULL);
+  /* Kick off an asynchronous save.
+   *
+   * FIXME: pass self->cancellable; see comment in
+   * epg_manager_shutdown_async().
+   */
+  g_assert_cmpuint (self->pending_internal_save_state_calls, <, G_MAXUINT64);
```

Comment:
Use `g_assert()`, rather than `g_assert_*()`, in non-test code. `g_assert_*()` log through the test logging code, and are affected by various test settings. `g_assert()` is disableable by compiling with `G_DISABLE_ASSERT`, which people typically want for higher performance production code. It also isn’t affected by any of the test settings.